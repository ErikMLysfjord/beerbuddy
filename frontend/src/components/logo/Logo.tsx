import styles from "./Logo.module.css";
import { CSSProperties } from "react";

const Logo = (props: { style?: CSSProperties | undefined }) => {
  return (
    <a style={props.style} className={styles.wrapper} href="/project2">
      <img
        src="/project2/teku.svg"
        className={styles.icon}
        alt="BeerBuddy logo"
      />
      <h1 className={styles.heading}>BeerBuddy</h1>
    </a>
  );
};
export default Logo;
