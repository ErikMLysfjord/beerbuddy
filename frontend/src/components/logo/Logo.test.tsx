import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Logo from "./Logo";
import { axe } from "jest-axe";

describe("Logo", () => {
  it("is accessible", async () => {
    const { container } = render(<Logo />);
    expect(await axe(container)).toHaveNoViolations();
  });

  it("renders correctly", () => {
    const { container } = render(<Logo />);
    expect(container).toMatchSnapshot();
  });

  it("renders with correct logo", () => {
    const { getByRole } = render(<Logo />);
    expect(getByRole("img")).toBeInTheDocument();
    expect(getByRole("img")).toHaveAttribute("alt", "BeerBuddy logo");
  });

  it("renders with correct link", () => {
    const { getByRole } = render(<Logo />);
    expect(getByRole("link")).toBeInTheDocument();
    expect(getByRole("link")).toHaveAttribute("href", "/project2");
  });

  it("renders with correct style", () => {
    const { getByRole } = render(<Logo style={{ color: "red" }} />);
    expect(getByRole("link")).toHaveStyle("color: rgb(255, 0, 0)");
  });
});
