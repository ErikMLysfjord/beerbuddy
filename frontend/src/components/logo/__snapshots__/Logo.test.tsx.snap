// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo > renders correctly 1`] = `
<div>
  <a
    class="_wrapper_4f3c83"
    href="/project2"
  >
    <img
      alt="BeerBuddy logo"
      class="_icon_4f3c83"
      src="/project2/teku.svg"
    />
    <h1
      class="_heading_4f3c83"
    >
      BeerBuddy
    </h1>
  </a>
</div>
`;
